pkg/statuscodes/statuscodes.go

```diff
@@ -47,6 +47,10 @@ const (
 	// are flooding the server.  It is expected that the client will back off for some duration and then try again.
 	// Well-behaved services will even return an expected duration for the client to retry-after.
 	RateLimited StatusCode = 705
+	// ClientCanceled is for when the client has canceled the request, and the server has not yet started processing.
+	// It is expectedc that sometimes the client will simply have a canceled request and no server request will have
+	// been made.
```

Comment:
```suggestion
	// ClientCanceled is for when the client has canceled the request, and the server has not yet started processing.
	// It is expected that sometimes the client will simply have a canceled request and no server request will have
	// been made.
```

Also, "the server has not yet started processing" isn't entirely correct, right? A lot of time the client might close the connection mid-request.

We might also want to document the intended usage of this status code since it will be a little difficult to understand for most people to determine how to use this (since the client has hanged up at this point, so how does one respond? for example.)

Reply:
@jaredallard I was cherry picking the case that I ran into which is the context indicates that its done. In that case I ran into, I would expect that I would simply open a new connection and retry. 

In the case where part of the request actually occurred and was cut off, I'm honestly not sure what would need to be done there. I can't think of a general action that could be take, nor how it would be figured out how to recover. WDYT

Reply:
Actually the more I think about it, the more I would just surface the error to the client with that grpc status code and they can choose their next step.